src/api/EscargotPublic.cpp

```diff
@@ -1297,6 +1297,158 @@ void DebuggerOperationsRef::BreakpointOperations::getLexicalScopeChain(uint32_t
     }
 }
 
+static void fillObjectProperties(ExecutionState* state, Object* object, Object::OwnPropertyKeyVector& keys, DebuggerOperationsRef::PropertyKeyValueVector& result, size_t start)
+{
+    size_t size = keys.size();
+
+    for (size_t i = 0; i < size; i++) {
+        ObjectPropertyName propertyName(*state, keys[i]);
+
+        result[start + i].key = toRef(keys[i].toStringWithoutException(*state));
```

Comment:
How about adding an assert statement to check that result vector has valid size?